app/main.py

```diff
@@ -1 +1,24 @@
-# write your code here
+from app.errors import (NotVaccinatedError,
+                        OutdatedVaccineError,
+                        NotWearingMaskError)
+
+
+def go_to_cafe(friends: list, cafe: str) -> str:
+    vaccinated_friends = 0
+    masks_to_buy = 0
+
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+            vaccinated_friends += 1
+        except NotVaccinatedError:
+            return "All friends should be vaccinated"
+        except OutdatedVaccineError:
+            return "All friends should be vaccinated"
+        except NotWearingMaskError:
+            masks_to_buy += 1
+
+    if vaccinated_friends == len(friends):
+        return f"Friends can go to {cafe.name}"
```

Comment:
you can omit this checking and  return the final message at the end of the function if none of the error conditions are met.